### ISSUE TITLE
Filters do not work in Graph Explorer

### ISSUE THREAD

**Issue by MAlabaster** [microsoftgraph/microsoft-graph-docs#15922]
If I paste this example of yours
GET https://graph.microsoft.com/v1.0/me/messages?$filter=subject eq 'let''s meet for lunch?'
into Graph Explorer, the textbox  shows invalid whitespace in URL even if I substitute %20 for spaces so I am at a total loss as to how to determine a filter clause let alone how to construct one in the C# SDK

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 44f6304a-ab90-8543-39c7-2afd61197ffb
* Version Independent ID: 9ca019b5-063f-33a2-3cb9-3ab6baafe9d4
* Content: [Use query parameters to customize responses - Microsoft Graph](https://docs.microsoft.com/en-us/graph/query-parameters)
* Content Source: [concepts/query-parameters.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/main/concepts/query-parameters.md)
* Product: **non-product-specific**
* Technology: **microsoft-graph**
* GitHub Login: @mumbi-o
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by lramosvea** [comment]
@RabebOthmani any idea about this?

**Comment by MAlabaster** [comment]
I managed to get the C# SDK to work using

filterClause = "contains(subject, 'let''s meet for lunch?')

theFolderMessages = await graphClient.Me.MailFolders[selectedFolderId].Messages
                .Request()
                .Top(500)
                .Filter(filterClause)
                .GetAsync();

but the GET statemet still eludes me

**Comment by ajh0912** [comment]
@MAlabaster So for me this Graph Explorer URI worked:
`GET` `https://graph.microsoft.com/v1.0/me/messages?$filter=subject eq 'let''s meet for lunch?'`

Although trying to call the same via Invoke-MgGraphRequest was a bit more awkward, had to percent encode the `?`:
```
Import-Module -Name Microsoft.Graph.Authentication -ErrorAction Stop
Connect-MgGraph -Scopes Mail.Read
(Invoke-MgGraphRequest -Uri "v1.0/me/messages?`$filter=subject eq 'let''s meet for lunch%3F'").value
```

And with the `Get-MgUserMessage` cmdlet:
```
Import-Module -Name Microsoft.Graph.Mail -ErrorAction Stop
Connect-MgGraph -Scopes Mail.Read
Get-MgUserMessage -UserId <UserId or UserPrincipalName> -Filter "subject eq 'let''s meet for lunch?'"
```

**Comment by MAlabaster** [comment]
I guess the issue is really one of documentation, there are a lot of parameters which are strings where it is very hard to discover what should be used and there are differences between the web, JSON and SDK formats. I had to pass a user agent string in Graph .net SDK calls to read a customer's emails because they have used EwsApplicationAccessPolicy as a "security" measure, it took me ages to work out that a) I had to use a header option in the Graph .Request() call  b) the header option name to use was "User-Agent". I was trying to use the Graph Explorer as a quicker way of experimenting than recompiling every time and it didn't work.
